youtrack_sdk/types.py

```diff
@@ -1,30 +1,64 @@
-from datetime import datetime
+from datetime import UTC, date, datetime
 from enum import StrEnum
+from typing import Any
 
-from pydantic.v1.datetime_parse import from_unix_seconds
+from pydantic._internal import _annotated_handlers
+from pydantic_core import core_schema
 
-from .exceptions import IncompatibleFieldTypeError, StrictIntError
+from .exceptions import StrictIntError
+from .helpers import from_unix_seconds
 
 
-class DateTime(datetime):
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, value, values):
-        if (project_custom_field := values.get("project_custom_field")) and (
-            project_custom_field.field.field_type.id != "date and time"
-        ):
-            raise IncompatibleFieldTypeError
+class CustomDate:
```

Comment:
So this is also extremely fishy.

You can see that for real `date` fields in YouTrack the timestamp looks like this `1690200000000`. This is because of the following:

https://www.jetbrains.com/help/youtrack/devportal/api-entity-DateIssueCustomField.html

> The value assigned to the custom field in the issue. The date is presented by the timestamp that **designates the midday of this date in UTC+0 timezone**. Can be null.

Therefore, this problem should be solved by an annotation that simply subtracts 12 hours from the timestamp.

https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.BeforeValidator

```python
UnixMiddayDate = Annotated[date, BeforeValidator(lambda d: d - 12 * 60 * 60 * 1000 if isinstance(d, int) else d)]
```

Reply:
@andruhamax you might be interested how people in the wild approach the "time doesn't make sense for a date" thing: new variation - let's take 12:00 UTC, this way the chance that it will be transformed wrong is minimized.

Reply:
Decision: we agreed to pursue this way in the meeting.